inc/actions.php

```diff
@@ -292,26 +289,14 @@ function uploadcare_settings_actions() {
  * Calculate final image dimensions
  */
 function uploadcare_get_final_dim($file) {
-    $resDim = array('width' => $file->data['image_info']->width,
-                         'height' => $file->data['image_info']->height);
-    $effectsStr = $file->default_effects;
-    $effectsArr = explode("/-/", $effectsStr);
-    foreach ($effectsArr as $key => $value) {
-        $effectParamArr = explode("/", $value);
-        switch($effectParamArr[0]) {
-            case "crop":
-                $tmpDimArr = explode('x', $effectParamArr[1]);
-                $resDim = array('width' => $tmpDimArr[0],
-                                'height' => $tmpDimArr[1]);
-            break;
-            case 'rotate':
-                if($effectParamArr[1] == '90' || $effectParamArr[1] == '270') {
-                    $resDim = array('width' => $resDim['height'],
-                                    'height' => $resDim['width']);
-                }
-            break;
-            default:
-        }
-    }
-    return $resDim;
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $file->getUrl().'-/json/');
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+        'Content-Type: application/json',
+        'Accept: application/json'
+    ));
+    $res = json_decode(curl_exec($ch), true);
+    curl_close($ch);
+    return array('width' => $res['width'], 'height' => $res['height']);
```

Comment:
will raise if there is no data in json. probably want to catch this?

Reply:
It's works fine, I did try on no image files. It's fine in PHP if you try to get value from array using key which does not exists, its returns NULL then.